sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/context/UserIdContextTest.java

```diff
@@ -56,4 +94,60 @@ public void prefixedInvalidUserId() {
     public void prefixedValidUserId() {
         assertEquals("c:alice", UserIdContext.getPrefixedUserId("c:alice"));
     }
+
+    @Test
+    public void setUserIdUserEquals() {
+        String mockUserId = "userId";
+        UserIdContext userIdContext = new UserIdContext();
+        UserIdContext.Listener listener = mock(UserIdContext.Listener.class);
+        userIdContext.addListener(listener);
+        userIdContext.setUserId(mockUserId);
+        verify(listener).onNewUserId(mockUserId);
+        assertEquals(userIdContext.getUserId(), mockUserId);
+        userIdContext.setUserId(mockUserId);
+        verify(listener).onNewUserId(mockUserId);
+        assertEquals(userIdContext.getUserId(), mockUserId);
+        userIdContext.setUserId(null);
+        verify(listener).onNewUserId(isNull(String.class));
+        assertNull(userIdContext.getUserId());
+    }
+
+    @Test
+    public void setUserIdUserNotEquals() {
+        String mockUserId1 = "userId1";
+        String mockUserId2 = "userId2";
+        UserIdContext userIdContext = new UserIdContext();
+        UserIdContext.Listener listener = mock(UserIdContext.Listener.class);
+        userIdContext.addListener(listener);
+        userIdContext.setUserId(mockUserId1);
+        verify(listener).onNewUserId(mockUserId1);
+        assertEquals(userIdContext.getUserId(), mockUserId1);
+        userIdContext.setUserId(mockUserId2);
+        verify(listener).onNewUserId(mockUserId2);
+        assertEquals(userIdContext.getUserId(), mockUserId2);
+        userIdContext.setUserId(null);
+        verify(listener).onNewUserId(isNull(String.class));
+        assertNull(userIdContext.getUserId());
+    }
+
+    @Test
+    public void addAndRemoveListeners() {
+        String mockUserId = "userId";
+        UserIdContext userIdContext = new UserIdContext();
+        UserIdContext.Listener listener = mock(UserIdContext.Listener.class);
+        userIdContext.addListener(listener);
+        userIdContext.setUserId(mockUserId);
+        verify(listener).onNewUserId(mockUserId);
+        userIdContext.removeListener(listener);
```

Comment:
you need to set a new userId after removing listener and before verifying interactions again, otherwise the end of test tests nothing.